pennylane/collections.py

```diff
@@ -213,29 +215,49 @@ def sum(x):
     return apply(np.sum, x)
 
 
-def _get_dot_func(interface):
+def _get_dot_func(interface, x=None):
     """Helper function for :func:`~.dot` to determine
-    the correct dot product function depending on the QNodeCollection
-    interface"""
+    the correct dot product function depending on the QNodeCluster
```

Comment:
```suggestion
    the correct dot product function depending on the QNodeCollection
```

Reply:
:thinking:

Reply:
:thinking: weird! I had used find-all-replace-all previous

Reply:
Actually, I think this happened due to a merge conflict.